Upgrade external tools

grihabor opened the pull request:
The pr was generated with this command:
```
pants run build-support/bin:external-tool-upgrade -- -v src/python/pants/backend/
```
(Notes are not generated yet)

The tool code is here https://github.com/pantsbuild/pants/pull/22064

huonw commented on docs/notes/2.26.x.md:
```diff
@@ -55,6 +55,20 @@ Pants has a new mechanism for `@rule` invocation in backends. In this release th
+#### Tool upgrades
+
+
+| Tool                           | Old version | New version |
+| ------------------------------ | ----------- | ----------- |
+| Buildifier                     | "7.1.2"     | "8.0.3"     |
+| GolangciLint                   | "1.51.1"    | "1.64.6"    |
+| HelmUnitTestSubsystem:         | "0.3.3"     | "0.8.0"     |
+| KubeconformSubsystem           | "0.6.2"     | "0.6.7"     |
+| NfpmSubsystem                  | "2.38.0"    | "2.41.3"    |
+| PyenvPythonProviderSubsystem   | "2.4.7"     | "2.5.3"     |
+| Ruff                           | "0.9.6"     | "0.9.10"    |
+| Taplo                          | "0.8.0"     | "0.9.3"     |
```
Minor, but maybe we can be more consistent with the notes here, in particular, referring to the tool's subsystem by the path the user will import it:

```suggestion
| Tool                           | Old version | New version |
| ------------------------------ | ----------- | ----------- |
| buildifier (`pants.backend.build_files.fmt.buildifier`)                     | "7.1.2"     | "8.0.3"     |
| golangci-lint (`pants.backend.experimental.go.lint.golangci_lint`)                   | "1.51.1"    | "1.64.6"    |
| helm-unittest (`pants.backend.experimental.helm`)         | "0.3.3"     | "0.8.0"     |
| Kubeconform (`pants.backend.experimental.helm.check.kubeconfirm`)           | "0.6.2"     | "0.6.7"     |
| nFPM (`pants.backend.experimental.nfpm`)                  | "2.38.0"    | "2.41.3"    |
| pyenv (`pants.backend.python.providers.experimental.pyenv`)   | "2.4.7"     | "2.5.3"     |
| Ruff (`pants.backend.experimental.python.lint.ruff.check` and `...ruff.format`)                           | "0.9.6"     | "0.9.10"    |
| Taplo (`pants.backend.tools.taplo`)                           | "0.8.0"     | "0.9.3"     |
```

huonw reviewed (approved):
Thank you!

grihabor commented:
@huonw Thank you for review! Ready for merge
- fixed merge conflict in pex-cli
- upgraded pex-cli to 2.33.4
- formatted the table in notes
- added a note about version format breaking change

grihabor commented:
> native_engine.IntrinsicError: Could not identify a process to backtrack to for: Missing digest: Was not present in the local store: Digest { hash: Fingerprint<ff1d94281acb95bbc53b5f4f2629a88e7afe6583dbd8f4b2941ae2e83895f80f>, size_bytes: 1294 }, with workunit: Workunit { name: "process", level: Debug, span_id: SpanId(3050899624272423132), parent_ids: [SpanId(10285682066423053561)], state: Started { start_time: SystemTime { tv_sec: 1741864346, tv_nsec: 218582000 }, blocked: false }, metadata: Some(WorkunitMetadata { desc: Some("Scheduling: Resolving plugins: hdrhistogram"), message: None, stdout: None, stderr: None, artifacts: [], user_metadata: [] }) }

I believe it's the flaky test again

grihabor commented on src/python/pants/backend/python/util_rules/pex_cli.py:
```diff
@@ -42,8 +41,8 @@ class PexCli(TemplatedExternalTool):
-    default_version = "v2.33.2"
-    default_url_template = "https://github.com/pex-tool/pex/releases/download/{version}/pex"
+    default_version = "2.33.4"
+    default_url_template = "https://github.com/pex-tool/pex/releases/download/v{version}/pex"
```
Sure https://github.com/pantsbuild/pants/pull/22065/commits/28dd523610d2b2c4b69ba90a89426a709054d6d0

Moved it to a separate pr https://github.com/pantsbuild/pants/pull/22090

huonw commented on docs/notes/2.26.x.md:
```diff
@@ -55,6 +55,21 @@ Pants has a new mechanism for `@rule` invocation in backends. In this release th
 - [yapf](https://www.pantsbuild.org/stable/reference/subsystems/yapf)
 
+#### Tool upgrades
+
```
Ah, sorry, for the dribble of a review (I could've flagged earlier), but let's provide a bit more context:

```suggestion

The default versions of various external tools have been upgraded. The previous default versions are still supported, by pinning using the `version = "..."` field within the subsystem in `pants.toml`.

```

grihabor replied:
Sure

huonw reviewed (approved):
Looks great now, except for some extra release notes. Thanks!

huonw commented:
Thank you!
